final-exam.html

```
@@ -0,0 +1,537 @@
+        </div>
+    </section>
+
+    <section class="challenge">
+        <h2>❌ Challenge 4: The Failing Test (Module 4)</h2>
+        <p>A test uses the wrong assertion, causing it to pass even when data types are different. Task: Change the assertion to a <strong>Strict Assertion</strong> as per our RFC 2119 "SHOULD" requirement.</p>
+        <p><strong>Task:</strong> Switch <code>assertEquals</code> to the strict type-safe version.</p>
+        <div class="code-block broken">
+            <pre><code>// BROKEN CODE
+public function testTypeSafety(): void {
+    $result = "100"; // String
+    $this->assertEquals(104, $result); // This might pass loosely in some versions!
```

HTMLHint (1.9.2)

[error] Special characters must be escaped : [ > ].

(spec-char-escape)